### PR TITLE
Correct `bundle show` deprecation

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
### What was the end-user problem that led to this PR?

The problem was `bundle show` deprecation messages are incorrect:

```
$ bundle show yard
[DEPRECATED FOR 2.0] use `bundle list` instead of `bundle show`
Resolving dependencies...
/home/deivid/Code/activeadmin/.bundle/ruby/2.5.0/gems/yard-0.9.16
$ bundle list yard
ERROR: "bundle list" was called with arguments ["yard"]
Usage: "bundle list"
```

### What was your diagnosis of the problem?

My diagnosis was that deprecation messages only mention `bundle list`, but in some cases, the replacement is `bundle info`.

### What is your fix for the problem, implemented in this PR?

My fix is to replace "show" in the original command with the appropriate alternative in each case.

### Why did you choose this fix out of the possible options?

I chose this fix because it was the most user friendly message, since it prints the exact command the user needs to type to get rid of the warning.

### CODE REVIEW

File: spec/bundler/cli_spec.rb [hunk 1]
```diff
@@ -71,17 +71,17 @@
     it "prints the running command" do
       gemfile ""
       bundle! "info bundler", :verbose => true
-      expect(last_command.stdout).to start_with("Running `bundle info bundler --no-color --verbose` with bundler #{Bundler::VERSION}")
+      expect(last_command.stdout).to start_with("Running `bundle info bundler --verbose` with bundler #{Bundler::VERSION}")
```
Comment (segiddins) [inline spec/bundler/cli_spec.rb]: why did this change?

File: spec/support/helpers.rb [hunk 2]
```diff
@@ -99,9 +99,6 @@ def bundle(cmd, options = {})
       with_sudo = options.delete(:sudo)
       sudo = with_sudo == :preserve_env ? "sudo -E" : "sudo" if with_sudo
 
-      no_color = options.delete("no-color") { cmd.to_s !~ /\A(e|ex|exe|exec|conf|confi|config)(\s|\z)/ }
-      options["no-color"] = true if no_color
```
Comment (segiddins) [inline spec/support/helpers.rb]: Why did you remove this?

File: spec/bundler/cli_spec.rb (hunk 1 above)
Comment (deivid-rodriguez) [reply spec/bundler/cli_spec.rb]: See below.

File: spec/support/helpers.rb (hunk 2 above)
Comment (deivid-rodriguez) [reply spec/support/helpers.rb]: I was trying to test the command line arguments that `bundle show` receives and for some reason I was getting the `--no-color` flag added to them? Then I found this flag and I added `s|sh|sho|show` as exceptions here and it worked.

But then I decided that doing this didn't make any sense to me, so I removed it so I could get some feedback from you :smiley:. Why are we doing that?

Comment (segiddins) [reply spec/support/helpers.rb]: So all the test output will never contain ANSI escape codes , and thus be consistent no matter how the tests are being run

Comment (deivid-rodriguez) [reply spec/support/helpers.rb]: Can you show an example of this inconsistency? Our current test environment seems to handle this just fine? (I think the explanation of why this just works is that an `open3` context is recognized by ruby as a "non-tty" situation, so it's automatically [turned off](https://github.com/bundler/bundler/blob/e115a839123a21365904fd86980452d2b2b0283e/lib/bundler/ui/shell.rb#L13) by `bundler`).

To me doing this is not good because the helper does not represent real world usage, but a slight modification of it, and it leads to weird assertions (like the ones I'm fixing here), and suprising behavior for the developer (when I first saw this I was like "what? where did this flag come from?", and I had to spend a while tracking it down and adding yet more exceptions here).

Comment (segiddins) [reply spec/support/helpers.rb]: You could also pass no-color => false to the bundle helper

Comment (deivid-rodriguez) [reply spec/support/helpers.rb]: Right. So is that what you're asking me to do?

Comment (deivid-rodriguez) [reply spec/support/helpers.rb]: I'm going to do that and make my case to remove this separately :slightly_smiling_face:.

Comment (deivid-rodriguez) [reply spec/support/helpers.rb]: This flag will be needed on most `bundle show` specs. Do you prefer that I explicitly pass the flag to every `bundle` invocation that needs it, that I add more exceptions inside the method, or that I leave this patch as it is?

Comment (segiddins) [reply spec/support/helpers.rb]: It's your call :) If it doesn't break the tests, I suppose removing it is fine?

Comment (deivid-rodriguez) [reply spec/support/helpers.rb]: Ok, yeah, I think it's fine!

**Comment by deivid-rodriguez** [comment]
@bundlerbot r+
